npm/private/utils.bzl

```diff
@@ -132,11 +132,6 @@ def _convert_v6_packages(packages):
             dependencies[dep_name] = _convert_pnpm_v6_package_name(dep_version)
         package_info["dependencies"] = dependencies
 
-        # optionalDependencies
```

Comment:
FWICT `optionalDependencies` are only in the `importers`, not here in the `packages`

Reply:
Hmm. I'm not sure about that since importers only deals with direct dependencies and not optional dependencies of transitives. Lets hold on this one for now.

Reply:
This is the type of edge case I want in that pnpm_lockfiles e2e test...

Reply:
I've updated this PR to demonstrate this content in the lockfile, although doesn't really test it.